### PR TITLE
Front page data binding

### PR DESCRIPTION
**Pull request by ouchadam**
Makes the front page api call via anonymous oauth token and binds the response to the ui

- no pagination, caching etc
- pretty ugly, no package management yet

frontpage

### CODE REVIEW

File: data/src/main/java/com/ouchadam/loldr/data/Repository.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                );
+
+                posts.add(post);
+            }
+
+            return new Feed(posts);
+        }
+    }
+
+    public Observable<Feed> subreddit(String subreddit) {
```
Comment (ataulm) [inline data/src/main/java/com/ouchadam/loldr/data/Repository.java]: :+1: cool! can we query for a list of subreddits?

Comment (ouchadam) [inline data/src/main/java/com/ouchadam/loldr/data/Repository.java]: nope for a list of posts from a subreddit, maybe this name isn't clear enough

Comment (ouchadam) [inline data/src/main/java/com/ouchadam/loldr/data/Repository.java]: annoyingly the front page isn't a subreddit

Comment (ataulm) [inline data/src/main/java/com/ouchadam/loldr/data/Repository.java]: I mean, can't we list subreddits, and then the user can pick one of those, then we use `subreddit("subredditName")`?

**Comment by ataulm** [comment]
:x: errrr where's class marshallerfactory lol

**Comment by ouchadam** [comment]
hahaha mac git sux

File: data/src/main/java/com/ouchadam/loldr/data/Repository.java (hunk 1 above)
Comment (ouchadam) [inline data/src/main/java/com/ouchadam/loldr/data/Repository.java]: I think it's `/subreddits/default` https://www.reddit.com/dev/api#GET_subreddits_default

**Pull request merged**